cl/lib/search_index_utils.py

```diff
@@ -104,3 +110,57 @@ def get_parties_from_case_name_bankr(case_name: str) -> list[str]:
         if separator in case_name:
             return cleaned_case_name.split(separator, 1)
     return [cleaned_case_name]
+
+
+def check_bulk_indexing_exception(
+    errors: list[dict[str, Any]], exception: str
+) -> bool:
+    """Check for a specific exception type in bulk indexing errors.
+    :param errors: A list of dictionaries representing errors from a bulk
+    indexing operation.
+    :param exception: The exception type string to check for in the error
+    details.
+    :return: True if the specified exception is found in any of the error
+    dictionaries; otherwise, returns False.
+    """
+    for error in errors:
+        if error.get("update", {}).get("error", {}).get("type") == exception:
+            return True
+    return False
+
+
+def index_documents_in_bulk(documents_to_index: list[dict[str, Any]]) -> None:
+    """Index documents in Elasticsearch using the bulk API.
+
+    :param documents_to_index: A list of dictionaries representing the documents
+    to be indexed in bulk.
+    :return: None.
+    """
+
+    client = connections.get_connection(alias="no_retry_connection")
+    # Execute the bulk update
+    ids = [doc["_id"] for doc in documents_to_index]
+    try:
+        bulk(client, documents_to_index)
+    except BulkIndexError as exc:
+        # Catch any BulkIndexError exceptions to handle specific error message.
+        # If the error is a version conflict, raise a ConflictError for retrying it.
+        if check_bulk_indexing_exception(
+            exc.errors, "version_conflict_engine_exception"
+        ):
+            raise ConflictError(
+                "ConflictError indexing cites.",
+                "",
+                {"ids": ids},
+            )
+        if check_bulk_indexing_exception(
+            exc.errors, "document_missing_exception"
+        ):
+            missing_opinion = exc.errors[0].get("update", {}).get("_id", None)
+            logger.warning(
+                "Opinion with ID %s is not indexed in ES.", missing_opinion
+            )
```

Comment:
Why are we only looking at the first error? the errors array can have info about multiple documents in the bulk operation, right? Shouldn't we go through the whole array?